internal/services/mssql/mssql_database_resource.go

```diff
@@ -191,56 +203,53 @@ func resourceMsSqlDatabaseCreate(d *pluginsdk.ResourceData, meta interface{}) er
 
 		// Update the SKUs of any partner databases where deemed necessary
 		for _, partnerDatabase := range partnerDatabases {
-			partnerDatabaseId, err := parse.DatabaseID(*partnerDatabase.ID)
+			partnerDatabaseId, err := commonids.ParseDatabaseID(*partnerDatabase.Id)
 			if err != nil {
-				return fmt.Errorf("parsing ID for Replication Partner Database %q: %+v", *partnerDatabase.ID, err)
+				return fmt.Errorf("parsing ID for Replication Partner Database %q: %+v", *partnerDatabase.Id, err)
 			}
 
 			// See: https://docs.microsoft.com/en-us/azure/azure-sql/database/active-geo-replication-overview#configuring-secondary-database
-			if partnerDatabase.Sku != nil && partnerDatabase.Sku.Name != nil && helper.CompareDatabaseSkuServiceTiers(skuName.(string), *partnerDatabase.Sku.Name) {
-				future, err := client.Update(ctx, partnerDatabaseId.ResourceGroup, partnerDatabaseId.ServerName, partnerDatabaseId.Name, sql.DatabaseUpdate{
-					Sku: &sql.Sku{
-						Name: utils.String(skuName.(string)),
-					},
+			if partnerDatabase.Sku != nil && partnerDatabase.Sku.Name != "" && helper.CompareDatabaseSkuServiceTiers(skuName.(string), partnerDatabase.Sku.Name) {
+				err := client.UpdateThenPoll(ctx, pointer.From(partnerDatabaseId), databases.DatabaseUpdate{
+					Sku: pointer.To(databases.Sku{
+						Name: skuName.(string),
+					}),
 				})
```

Comment:
To match other parts of the provider can we split this onto multiple lines:

```suggestion
				updatePayload := databases.DatabaseUpdate{
					Sku: pointer.To(databases.Sku{
						Name: skuName.(string),
					}),
				}
				err := client.UpdateThenPoll(ctx, pointer.From(partnerDatabaseId), updatePayload)
```

Reply:
Fixed.